### PR TITLE
Add osg-ce-attributes-generator (SOFTWARE-4895)

### PR DESCRIPTION
**Pull request by matyasselmeci**
Standalone version of the code that makes /etc/condor-ce/config.d/10-osg-attributes-generated.conf.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -475,3 +475,79 @@ This section is contained in `/etc/osg/config.d/40-siteinfo.ini` which is provid
...
+Configuration
+-------------
+
+The CE Attributes Generator uses the same config files that OSG-Configure uses.
+See the [31-cluster.ini](#31-clusterini--subcluster-and-resource-entry-sections),
+[35-pilot.ini](#35-pilotini--pilot) for attributes.
+
+You will also need `resource` and `resource_group` from the `Site Information` section
+([40-siteinfo.ini](#40-siteinfoini--site-information-section)).
+You can also use `--resource` and `--resource-group` on the command line to specify these.
```
Comment (brianhlin) [inline README.md]:
```suggestion
You can also use `--resource` and `--resource-group` on the command line to specify these
instead.
```

File: README.md [hunk 2]
```diff
@@ -475,3 +475,79 @@ This section is contained in `/etc/osg/config.d/40-siteinfo.ini` which is provid
...
+
+In addition, you should also have at least one batch system section with the attribute `enabled=True`, e.g.
```
Comment (brianhlin) [inline README.md]:
```suggestion
In addition, you need at least one batch system section with the attribute `enabled=True`, e.g.
```

What happens if they have more than one batch system?

File: osg_configure/modules/ce_attributes.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+def get_resource_group_from_config(config: ConfigParser) -> str:
+    return utilities.classad_quote(
+        empty_if_blank(
+            config.get("Site Information", "resource_group", fallback="")
+        )
+    )
+
+
+def get_batch_systems_from_config(config: ConfigParser) -> str:
+    batch_systems = []
```
Comment (brianhlin) [inline osg_configure/modules/ce_attributes.py]: Should this be a set to avoid potential duplicates?

**Review by brianhlin — APPROVED** [review]
Looks reasonable to me. A few nitpicks and a question

File: README.md (hunk 2 above)
Comment (matyasselmeci) [reply README.md]: They don't need a batch system section with enabled=True if they pass --batch-systems on the command line, or if they're using BOSCO and specifying the batch system in `20-bosco.ini`.

If they have multiple batch systems, the `OSG_BatchSystems` attribute ends up being a comma-separated list.

File: osg_configure/modules/ce_attributes.py (hunk 3 above)
Comment (matyasselmeci) [reply osg_configure/modules/ce_attributes.py]: Yeah, sure.

File: README.md (hunk 2 above)
Comment (brianhlin) [reply README.md]: > They don't need a batch system section with enabled=True if they pass --batch-systems on the command line, or if they're using BOSCO and specifying the batch system in 20-bosco.ini.

Hrmm, neither the "should also have" or "need" versions of this sentence communicate that logic so we should probably number the different ways to specify the batch system in order of prio

**Pull request merged**